rust/src/storage/mod.rs

```diff
@@ -9,14 +9,48 @@
 // this software will be governed by the GNU Lesser General Public License v3.
 #![allow(dead_code)]
 
-use std::path::Path;
+use std::{fs::OpenOptions, path::Path};
 
 pub use self::error::Error;
 use crate::error::BTResult;
 
 mod error;
 pub mod file;
 pub mod storage_with_flush_buffer;
+#[cfg(test)]
+use tests::all_db_modes;
+
+/// The mode in which the database can be opened.
+#[derive(Debug, Clone, Copy)]
+pub enum DbMode {
+    Read,
```

Comment:
```suggestion
    ReadOnly,
```

It seems like in most places this is being referred to as "read only" already (e.g. the new error variant, or in test names). WDYT?